### PR TITLE
ENH: Add daily returns factor.

### PR DESCRIPTION
**Pull request by ssanderson**

### CODE REVIEW

File: tests/pipeline/test_factor.py [hunk 1]
```diff
@@ -1283,3 +1287,24 @@ class F(Factor):
             f.to_workspace_value(pipeline_output, pd.Index([0, 1])),
             column_data,
         )
+
+
+class TestSpecialCases(WithEquityPricingPipelineEngine,
+                       ZiplineTestCase):
+
+    def check_equivalent_terms(self, terms):
```
Comment (llllllllll) [inline tests/pipeline/test_factor.py]: should you just check that DailyReturns() is Return(window_length=2)?

**Comment by llllllllll** [comment]
one comment about testing, otherwise lgtm

File: tests/pipeline/test_factor.py (hunk 1 above)
Comment (ssanderson) [reply tests/pipeline/test_factor.py]: The `is` won't be true, b/c `DailyReturns` is a subclass.

Comment (ssanderson) [reply tests/pipeline/test_factor.py]: This would be an interesting use-case for graph rewrite rules if we ever add them.

**Pull request merged**